app/Http/Controllers/Api/UsersController.php

```diff
@@ -74,6 +83,17 @@ public function store(UserRequest $request)
         }
 
         $contest->waitingRoom->users()->attach($user->id);
+        $this->manager->appendCampaign($contest);
+
+        // Fire off welcome Email
+        $message = Message::where(['contest_id' => $contest->id, 'type' => 'welcome'])->first();
```

Comment:
I don't think the `Email` class should be responsible for firing off events because the event handler ends up instantiating the Email class to build the email object we need for actual sending. Also wary of it being the `Message` class since this needs to gather stuff from many different models. If anything this should be a function in the `Manger` that can be called in the controller.